### PR TITLE
Improve support for unioned queries on MySQL.

### PR DESCRIPTION
**Pull request by markstory**
By wrapping unioned queries with () each arm in an union can have an order clause. I've created a specialized compiler for SQLite as it seems to be the only rdbms that does not support () in UNION from what I could see.

Refs #5962

### CODE REVIEW

**Comment by lorenzo** [comment]
I think one of the tests is making sqlserver unhappy

File: tests/TestCase/Database/QueryTest.php [hunk 1]
```diff
@@ -1945,6 +1945,31 @@ public function testUnion()
...
+            ->order(['a.id' => 'asc']);
+
+        $query = new Query($this->connection);
+        $result = $query->select(['id', 'comment'])
+            ->from(['c' => 'comments'])
+            ->order(['c.id' => 'asc'])
+            ->union($union)
+            ->execute();
+        $this->assertCount(self::COMMENT_COUNT + self::ARTICLE_COUNT, $result);
+    }
```
Comment (lorenzo) [inline tests/TestCase/Database/QueryTest.php]: missing `closeCursor()` here, I think this is why SQLServer is whining

**Comment by markstory** [comment]
SQLServer like SQLite does not support `ORDER BY` in the queries used in an `UNION`. If you want to order a UNION in both you have to use `SELECT * FROM ( <union>) AS results ORDER BY results.column`. I'll exclude SQLserver from this new behavior as well.

**Comment by markstory** [comment]
AppVeyor and Travis passing now.

**Pull request merged**